integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,529 @@
+	lb, err := utils.NewLoadBalancer(context.Background(), *lbAddr)
+	require.NoError(t, err)
+
+	require.NoError(t, lb.Listen())
+	go lb.Serve()
+
+	p.lb = lb
+}
+
+// makeAuth bootsraps a new teleport auth instance.
+func (p *proxyTunnelStrategy) makeAuth(t *testing.T) {
+	if p.auth != nil {
+		require.Fail(t, "auth already initialized")
+	}
+
+	privateKey, publicKey, err := testauthority.New().GenerateKeyPair()
+	require.NoError(t, err)
+
+	auth := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		Priv:        privateKey,
+		Pub:         publicKey,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	auth.AddUser(p.username, []string{p.username})
+
+	conf := service.MakeDefaultConfig()
+	conf.DataDir = t.TempDir()
+	conf.Auth.Enabled = true
+	conf.Auth.NetworkingConfig.SetTunnelStrategy(p.strategy)
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+
+	require.NoError(t, auth.CreateEx(t, nil, conf))
+	require.NoError(t, auth.Start())
+
+	p.auth = auth
+}
+
+// makeProxy bootstraps a new teleport proxy instance.
+// It's public address points to a load balancer.
+func (p *proxyTunnelStrategy) makeProxy(t *testing.T) {
+	proxy := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	authAddr := utils.MustParseAddr(net.JoinHostPort(p.auth.Hostname, p.auth.GetPortAuth()))
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, *authAddr)
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.SSH.Enabled = false
+
+	conf.Proxy.Enabled = true
+	conf.Proxy.WebAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PeerAddr.Addr = net.JoinHostPort(Loopback, strconv.Itoa(ports.PopInt()))
+	conf.Proxy.PublicAddrs = append(conf.Proxy.PublicAddrs, utils.FromAddr(p.lb.Addr()))
+	conf.Proxy.DisableWebInterface = true
+
+	require.NoError(t, proxy.CreateEx(t, nil, conf))
+	p.lb.AddBackend(conf.Proxy.WebAddr)
+	require.NoError(t, proxy.Start())
+
+	p.proxies = append(p.proxies, proxy)
+}
+
+// makeNode bootstraps a new teleport node instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeNode(t *testing.T) {
+	if p.node != nil {
+		require.Fail(t, "node already initialized")
+	}
+
+	node := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = true
+
+	require.NoError(t, node.CreateEx(t, nil, conf))
```

Comment:
Same configuration caveat as the proxy one, `AuthServers` will contain the load balancer and the `node`'s internal (deactivated) auth server.